templates/mon_mysql.sql.j2

```diff
@@ -6,7 +6,7 @@ SET foreign_key_checks = 0;
 CREATE TABLE `alarm` (
   `id` varchar(36) COLLATE utf8mb4_unicode_ci NOT NULL,
   `alarm_definition_id` varchar(36) COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',
-  `state` enum('UNDETERMINED','OK','ALARM') COLLATE utf8mb4_unicode_ci NOT NULL,
```

Comment:
That was our idea as well but we decided that it'd be better to add triggers instead of adding new tables. Basically because of easier migration. 

Anyway we will investigate that, first thing Monday morning. That seems to be possible to map such schema in Hibernate because an enum (considered as Java type) can be mapped in two ways. For now we have a string, but there's an option to use a number that would be understood in schema as foreign key.